### DISCUSSION TITLE
Structured Configs with non-primitive types

### DISCUSSION THREAD

**Discussion by syed-ahmed** [facebookresearch/hydra#2138] [Q&A] [answered]
Hi,

I'm trying to create a config like this:
```python
@dataclass
class MatMulConfig:
    m: int = 10240
    n: int = 10240
    k: int = 10240
    dtype: torch.dtype = torch.double
```
and got this error:
```
    raise ValidationError(f"Unexpected object type: {type_str(type_)}")
omegaconf.errors.ValidationError: Unexpected object type: dtype
```
but then realized reading the docs that this is not possible. 

My use case is: `python my_app.py dtype=torch.float` where `torch.float` is a `torch.dtype` type. Is there a way to achieve this?

**Comment by Jasha10** [comment, marked as answer]
Currently I think there is no clean way to achieve this. You could use a look-up table in your code:
```python
@dataclass
class MatMulConfig:
    m: int = 10240
    n: int = 10240
    k: int = 10240
    dtype: str = "double"

dtype_lookup = {
  "double": torch.double,
  "float": torch.float,
  ...
}
dtype = dtype_lookup[cfg.dtype]
```

**Reply by Jasha10** [reply]
I've opened two feature requests that might help with this use-case in the future: #2139 and #2140.

**Reply by syed-ahmed** [reply]
Thanks @Jasha10!
